### PR TITLE
1376: Add CSS to current navigation menu item

### LINKED ISSUE #1376: Navigation menu doesn't show 'current' item
While navigating between **Projects**, **Locales**, **Profile** or **Settings** the navigation menu don't identifies the current item.

#### Current UI
imagem

#### Example adding `--gp-color-primary` as `background-color`:
imagem

### PR DESCRIPTION
**Pull request by pedro-mendonca**
This allows to clearly identify the active navigation menu item.

- [x] ~Change home `h1` to `nav#home-navigation` for better `:hover` consistency~
- [x] Set `.current` ~`background-color`~ underline color to `--gp-color-primary`
- [x] Set `:hover` ~`background-color`~ underline color to `--gp-color-primary-200`  (the color immediately above primary in the lighter spectrum)
- [x] Remove `.gp-bar` left and right padding to keep the navbar and content aligned

Update:
- [x] Don't change home `h1` link as reviewed
- [x] Use styling **Option 3** (see [options](https://github.com/GlotPress/GlotPress/pull/1379#issuecomment-1103797410) below) for styling
### Screenshot
Projects `.current` and Locales `:hover`

imagem

Fixes #1376

### CODE REVIEW

**Review by ocean90 — COMMENTED** [review]
I'd prefer something more subtle here. Maybe `box-shadow: inset 0 -5px 0 var( --gp-color-secondary-900 )` and change to white on hover instead of current text underline?

image

**Comment by pedro-mendonca** [comment]
@ocean90 it has too low contrast, hardly noted.
I agree on removing the underline.

The GlotPress title is wrapped in a `<h1>`, maybe should be removed too?

**Comment by pedro-mendonca** [comment]
Below are a few possible navbar solutions.

These examples have the **Projects** as the `.current` item, and the **Profile** is the `:hover`/`:focus` status.

1. The current PR status
imagem

2. The @ocean90 suggestion on [this comment](https://github.com/GlotPress/GlotPress/pull/1379#pullrequestreview-929119385)
imagem

3. A third option, using the below box-shadow but with higher contrast on `.current` and using the glotpress `--gp-color-primary` variations.
imagem


Please give feedback.

**Comment by amieiro** [comment]
In my opinion, the third option is the best: it is subtle, but it has more contrast than if you use gray or white.

File: gp-includes/template.php [hunk 1]
```diff
@@ -97,8 +97,15 @@ function gp_nav_menu( $location = 'main' ) {
 	$html  = '';
 	$items = gp_nav_menu_items( $location );
 
+	// Get the current URI.
+	global $wp;
+	$gp_route    = isset( $wp->query_vars['gp_route'] ) ? $wp->query_vars['gp_route'] : '';
+	$current_uri = gp_url( $gp_route );
```
Comment (ocean90) [inline gp-includes/template.php]: You can use `gp_url_current()` here.

File: gp-includes/template.php [hunk 2]
```diff
@@ -97,8 +97,15 @@ function gp_nav_menu( $location = 'main' ) {
...
+
 	foreach ( $items as $link => $title ) {
-		$html .= '<a href="' . $link . '">' . $title . '</a>';
+		// Check if the link matches the current URI base, if true, add 'current' class.
+		$class = str_starts_with( $current_uri, $link ) ? 'current' : '';
```
Comment (ocean90) [inline gp-includes/template.php]: `str_starts_with()` is a PHP 8 function and only polyfilled in WordPress 5.9+. You can use `gp_startswith()` instead.

File: gp-templates/header.php [hunk 3]
```diff
@@ -16,7 +16,7 @@
 	<script type="text/javascript">document.body.className = document.body.className.replace('no-js','js');</script>
 
 	<header class="gp-bar clearfix">
-		<h1>
+		<nav id="home-navigation">
```
Comment (ocean90) [inline gp-templates/header.php]: This is not a navigation, let's keep it as h1 for now.

**Review by ocean90 — CHANGES_REQUESTED** [review]
Using the primary color as an underline for the current item sounds good too. But there's no need to use different colors for the left and right menu, they can use the same colors to keep it simple.

**Comment by pedro-mendonca** [comment]
@ocean90 As I've mentioned above, the left and right colors show different status, on the left is a primary color to show the current status, and on the right I've placed a primary-color-200 to differentiate on hover on the current item.
Having a slight lighter color on hover allows is to have an hover effect also on current item.

**Comment by pedro-mendonca** [comment]
@ocean90 and @amieiro I've changed to the option 3 in https://github.com/GlotPress/GlotPress/pull/1379/commits/7b57892f0ae8292312169aa7f50b29b1484871a6

**Review by ocean90 — APPROVED** [review]

**Comment by pedro-mendonca** [comment]
Thanks for fixing the loose ends :)

**Pull request merged**